internal/chacha/chacha_arm64.go

```diff
@@ -0,0 +1,54 @@
+//go:build gc
+
+package chacha
+
+import "encoding/binary"
+
+func init() {
+	useVX = true
+}
+
+func initialize(state *[64]byte, key []byte, nonce *[16]byte) {
+	binary.LittleEndian.PutUint32(state[0:], sigma[0])
+	binary.LittleEndian.PutUint32(state[4:], sigma[1])
+	binary.LittleEndian.PutUint32(state[8:], sigma[2])
+	binary.LittleEndian.PutUint32(state[12:], sigma[3])
+	copy(state[16:], key[:])
+	copy(state[48:], nonce[:])
+}
+
+func xorKeyStream(dst, src []byte, block, state *[64]byte, rounds int) int {
+	switch {
+	case useVX:
```

Comment:
do we even need a switch case if `useVX` will always be true?

Reply:
This is implemented in the same vein as the similar flags on the package.

The variable can be changed in tests to ensure the version without assembly can be tested too.

Reply:
ah, makes sense